src/Common/src/System/Security/Cryptography/Asn1V2.cs

```diff
@@ -704,6 +707,120 @@ public override int GetMaxCharCount(int byteCount)
+
+        public override int GetByteCount(string s)
+        {
+            return s_utf8Encoding.GetByteCount(s);
+        }
+
+#if netcoreapp || uap
+        public override int GetByteCount(ReadOnlySpan<char> chars)
+        {
+            return s_utf8Encoding.GetByteCount(chars);
+        }
+#endif
+
+        public override int GetBytes(char[] chars, int charIndex, int charCount, byte[] bytes, int byteIndex)
+        {
+            return s_utf8Encoding.GetBytes(chars, charIndex, charCount, bytes, byteIndex);
+        }
+
+        public override unsafe int GetBytes(char* chars, int charCount, byte* bytes, int byteCount)
+        {
+            return s_utf8Encoding.GetBytes(chars, charCount, bytes, byteCount);
+        }
+
+        public override int GetCharCount(byte[] bytes, int index, int count)
+        {
+            try
+            {
+                return s_utf8Encoding.GetCharCount(bytes, index, count);
+            }
+            catch (DecoderFallbackException)
+            {
+                return s_latin1Encoding.GetCharCount(bytes, index, count);
+            }
+        }
+
+        public override unsafe int GetCharCount(byte* bytes, int count)
+        {
+            try
+            {
+                return s_utf8Encoding.GetCharCount(bytes, count);
+            }
+            catch (DecoderFallbackException)
+            {
+                return s_latin1Encoding.GetCharCount(bytes, count);
+            }
+        }
+
+#if netcoreapp || uap
+        public override int GetCharCount(ReadOnlySpan<byte> bytes)
+        {
+            try
+            {
+                return s_utf8Encoding.GetCharCount(bytes);
+            }
+            catch (DecoderFallbackException)
+            {
+                return s_latin1Encoding.GetCharCount(bytes);
+            }
+        }
+#endif
+
+        public override int GetChars(byte[] bytes, int byteIndex, int byteCount, char[] chars, int charIndex)
+        {
+            try
+            {
+                return s_utf8Encoding.GetChars(bytes, byteIndex, byteCount, chars, charIndex);
+            }
+            catch (DecoderFallbackException)
+            {
+                return s_latin1Encoding.GetChars(bytes, byteIndex, byteCount, chars, charIndex);
+            }
+        }
+
+        public override unsafe int GetChars(byte* bytes, int byteCount, char* chars, int charCount)
+        {
+            try
+            {
+                return s_utf8Encoding.GetChars(bytes, byteCount, chars, charCount);
+            }
+            catch (DecoderFallbackException)
+            {
+                return s_latin1Encoding.GetChars(bytes, byteCount, chars, charCount);
+            }
+        }
+
+        public override int GetMaxByteCount(int charCount)
+        {
+            return s_utf8Encoding.GetMaxByteCount(charCount);
+        }
+
+        public override int GetMaxCharCount(int byteCount)
+        {
+            return s_utf8Encoding.GetMaxCharCount(byteCount);
```

Comment:
This one is really just `return byteCount`, no?  With a comment along the lines of

* If the bytes are a valid UTF-8 sequence then the maximum UTF-8 char count, which is `byteCount` for 7-bit ASCII-capable payloads and smaller numbers otherwise.
* Otherwise, the Latin-1 char count, which is defined to be `byteCount`.

Really the problem is that it is the only byte->char method which doesn't talk about Latin-1, and then I'm optimizing the Math.Max call, and the two deferred calls.

Reply:
Yeah, it is just `return byteCount`. I wanted to put in a comment about Latin-1 being single-byte character encoding, but then I thought it may be too obvious. I will simplify it and add a comment.